util/src/main/java/org/hyperledger/besu/util/NetworkUtility.java

```diff
@@ -98,4 +105,31 @@ public static void checkPort(final int port, final String portTypeName) {
               "%s port requires a value between 1 and 65535. Got %d.", portTypeName, port));
     }
   }
+
+  public static boolean isPortAvailableForTcp(final int port) {
+    try (final ServerSocket serverSocket = new ServerSocket()) {
+      serverSocket.setReuseAddress(true);
+      serverSocket.bind(new InetSocketAddress(port));
+      return true;
+    } catch (IOException ex) {
+      LOG.trace(String.format("Failed to open port %d for TCP", port), ex);
+    }
+    return false;
+  }
+
+  public static boolean isPortAvailableForUdp(final int port) {
+    try (final DatagramSocket datagramSocket = new DatagramSocket(null)) {
+      datagramSocket.setReuseAddress(true);
+      datagramSocket.bind(new InetSocketAddress(port));
+      return true;
+    } catch (IOException ex) {
+      LOG.trace(String.format("failed to open port %d for UDP", port), ex);
+    }
+    return false;
+  }
+
+  public static boolean isPortAvailable(final int port) {
+    if (isPortAvailableForTcp(port) && isPortAvailableForUdp(port)) return true;
```

Comment:
Could simplify this to
`return isPortAvailableForTcp(port) && isPortAvailableForUdp(port)`

Reply:
Done, thanks!